src/jimgw/core/single_event/likelihood.py

```
@@ -379,6 +383,117 @@ def _likelihood(self, params: dict[str, Float], data: dict) -> Float:
+
+        if dist_prior is None:
+            raise ValueError(
+                "dist_prior must be provided. "
+                "Example: PowerLawPrior(xmin=100, xmax=5000, alpha=2, parameter_names=['d_L'])"
+            )
+
+        if not hasattr(dist_prior, "xmin") or not hasattr(dist_prior, "xmax"):
+            raise ValueError(
+                "dist_prior must have xmin and xmax attributes defining the distance bounds. "
+                "Use a bounded prior such as PowerLawPrior or UniformPrior."
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)